[JBTM-3506] removing maven binaries from source code and using maven wrapper instead

ochaloup opened the pull request:
https://issues.redhat.com/browse/JBTM-3506

MAIN QA_JTA XTS WIN BLACKTIE
 !QA_JTS_JDKORB !QA_JTS_OPENJDKORB !QA_JTS_JACORB !PERF !RTS !AS_TESTS !TOMCAT !JACOCO !LRA

mmusgrov reviewed (approved):
Perfect, and quick work.

tomjenkinson commented on .mvn/wrapper/settings.xml:
```diff
@@ -255,6 +255,15 @@ under the License.
           <url>https://repo1.maven.org/maven2</url>
         </repository>
       </repositories>
+      <pluginRepositories>
```
@ochaloup if this was not something we were doing before and relates to downloads over HTTP vs HTTPS then I think we should have it in a separate commit

ochaloup replied:
@tomjenkinson ok. do I need a separate JIRA for this or is it ok to just mention it in the commit description?

mmusgrov commented:
Also I noticed that some of the readme files still refer to mvn.

ochaloup commented:
@mmusgrov I updated the PR with changes in the README while rebased on top of the current master
@tomjenkinson I didn't realize the fact about the blacktie. I adjusted the PR and tried to run the xml pkg on my local machine and it works. I'm adding the BLACKTIE axis to be check for this PR and will see if I understand the code correctly to pass on CI as well. I added a commit with the plugin repositories on top of the original changeset and not within it.

ochaloup commented:
I'm going to merge this and I hope that all will work fine. Let me know if not :-)
